numpy/lib/arraysetops.py

```diff
@@ -268,6 +268,39 @@ def unique(ar, return_index=False, return_inverse=False,
         # this removes the "axis1" or "axis2" prefix from the error message
         raise np.AxisError(axis, ar.ndim)
 
+    if ar.size == 0:
+        # if there's no elements (at least one axis is 0), the reshaping and
+        # viewing doesn't work, but there's also no data, so those
+        # manipulations aren't needed to get right answer.
+
+        axis_len = ar.shape[0]
+        if axis_len == 0:
+            # there's no elements along this axis, so there's no unique
+            # elements:
+            num_empties = return_index + return_inverse + return_counts
+            out = ar
+            extra_output = (np.array([], dtype=np.intp),) * num_empties
+        else:
+            # there are elements along this axis. Each element is empty, so
+            # there's a single unique example of this element
+            output_shape = list(ar.shape)
+            output_shape[0] = 1
+            out = np.empty(shape=output_shape, dtype=ar.dtype)
+
+            extra_output = ()
+            if return_index:
+                # first index is the first occurance of an empty element
```

Comment:
Spelling: occurrence

Reply:
I fixed the typo, but #15747 looks better.